### ISSUE TITLE
MAJOR: Need to remove boto session caching and replace with credential caching

### ISSUE THREAD

**Issue by mikegrima** [Netflix-Skunkworks/cloudaux#58]
The boto session caching is causing massive memory leaks with newer versions of botocore.

Will need to only cache credentials instead.

**Comment by 0xdabbad00** [comment]
I might be seeing something similar in a different project with botocore.  Do you have any links to this issue?

**Comment by mikegrima** [comment]
Hello @0xdabbad00 : Nothing explicitly linked anywhere. We began noticing some major issues with Security Monkey, where it was effectively crippled. 

To resolve, I made a very significant (and very long overdue) refactor to make use of Celery and to segment all watchers into individual processes, and to have those processes terminate when complete to free up available RAM.

We noticed on our other major project [Netflix-Skunkworks/Historical](https://github.com/Netflix-Skunkworks/historical), that we observed similar memory leakage.  @kevgliss did some major investigative work, and tracked down that some where after botocore version 1.7.45(IIRC?), there has been major memory leakage when retaining onto old clients.

Our attempt to permanently resolve this issue is to cache the credentials instead of the boto connections, and to simply create new botocore sessions with the cached credentials each time. We are hopeful that this would help resolve the issue.
